lib/controller.js

```diff
@@ -457,6 +460,167 @@ controller.connectToNetwork = function(opts) {
     });
 };
 
+controller.update = function(opts) {
+  return new Promise(function(resolve, reject) {
+    // If the user requested a list of builds
+    if (opts.list) {
+      // Fetch those builds from the build server
+      logs.info('Fetching available builds...');
+      return updates.fetchBuildList()
+        .then(function(builds) {
+          logs.info('Latest builds:');
+          var maxBuildsListed = 10;
+          var SHAs = Object.keys(builds);
+
+          // Limited the number of builds we print out
+          SHAs = SHAs.splice(0, SHAs.length > maxBuildsListed ? maxBuildsListed : SHAs.length);
+          // Print out details for each build
+          for (var i = 0; i < SHAs.length; i++) {
+            // Grab the specific build
+            var build = builds[SHAs[i]];
+            // Format the release time as a date
+            var released = new Date(build.released);
+            // Print when it was published
+            logs.basic('\t Version:', build.version, '\tPublished:', released.toLocaleString());
+          }
+          // Finish up
+          return resolve();
+        }, reject);
```

Comment:
With my suggested change to the builds.json file, this whole operation (fetch and operation handler body) can be replaced with: 

```js
return updates.fetchBuildList()
  .then(function(builds) {
    logs.info('Latest builds:');
    
    builds.slice(-10).forEach(function(build) {
      logs.basic('\t Version:', build.version, '\tPublished:', released.toLocaleString());
    });
    // Finish up
    return resolve();
  }, reject);
```